contribs/src/main/java/com/netflix/conductor/contribs/queue/amqp/AMQPObservableQueue.java

```diff
@@ -366,27 +359,23 @@ public AMQPObservableQueue build(final boolean useExchange, final String queueUR
     }
 
     private Channel getOrCreateChannel() {
-        if (!isConnOpened) {
-            open();
-        }
-        // Return the existing channel if it's still opened
-        if (channel != null && isChanOpened) {
+        // Return the existing channel if it was created
+        if (channel != null) {
             return channel;
         }
         // Channel creation is required
         try {
-            channel = null;
+            connect();
             channel = connection.createChannel();
             channel.addShutdownListener(cause -> {
-                isChanOpened = false;
```

Comment:
Would it make sense to retry channel creation in shutdown listener after a configurable time interval?